src/SAWScript/Crucible/LLVM/ResolveSetupValue.hs

```diff
@@ -356,6 +358,28 @@ resolveSAWPred ::
 resolveSAWPred cc tm =
   Crucible.bindSAWTerm (cc^.ccBackend) W4.BaseBoolRepr tm
 
+resolveSAWSymBV ::
+  (1 <= w) =>
+  LLVMCrucibleContext arch ->
+  NatRepr w ->
+  Term ->
+  IO (W4.SymBV Sym w)
+resolveSAWSymBV cc w tm =
+  do let sym = cc^.ccBackend
+     sc <- Crucible.saw_ctx <$> readIORef (W4.sbStateManager sym)
+     let ss = cc^.ccBasicSS
+     tm' <- rewriteSharedTerm sc ss tm
+     mx <- case getAllExts tm' of
+             [] -> do
+               -- Evaluate in SBV to test whether 'tm' is a concrete value
+               modmap <- scGetModuleMap sc
+               sbv <- SBV.toWord =<< SBV.sbvSolveBasic modmap Map.empty [] tm'
```

Comment:
Could we use What4 instead of SBV here? Our goal for the release after next is to deprecate all functions that depend on SBV, as long as there aren't any serious performance regressions from doing so.

Reply:
in general, it's possible. Here, I just extracted existing code from `resolveSAWTerm` into a new function `resolveSAWSymBV` in order to reuse code. I prefer to not mix the SBV -> What4 update in this code change.

Reply:
That makes sense. Perhaps we should open an issue to track a later change.